File: software/examples/vbnet/ExampleCallback.vb
Imports System
Imports Tinkerforge

Module ExampleCallback
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Multi Touch Bricklet 2.0

    ' Callback subroutine for touch state callback
    Sub TouchStateCB(ByVal sender As BrickletMultiTouchV2, ByVal state As Boolean())
        Console.WriteLine("Electrode 0: " + state(0).ToString())
        Console.WriteLine("Electrode 1: " + state(1).ToString())
        Console.WriteLine("Electrode 2: " + state(2).ToString())
        Console.WriteLine("Electrode 3: " + state(3).ToString())
        Console.WriteLine("Electrode 4: " + state(4).ToString())
        Console.WriteLine("Electrode 5: " + state(5).ToString())
        Console.WriteLine("Electrode 6: " + state(6).ToString())
        Console.WriteLine("Electrode 7: " + state(7).ToString())
        Console.WriteLine("Electrode 8: " + state(8).ToString())
        Console.WriteLine("Electrode 9: " + state(9).ToString())
        Console.WriteLine("Electrode 10: " + state(10).ToString())
        Console.WriteLine("Electrode 11: " + state(11).ToString())
        Console.WriteLine("Proximity: " + state(12).ToString())
        Console.WriteLine("")
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim mt As New BrickletMultiTouchV2(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Register touch state callback to subroutine TouchStateCB
        AddHandler mt.TouchStateCallback, AddressOf TouchStateCB

        ' Set period for touch state callback to 0.01s (10ms)
        mt.SetTouchStateCallbackConfiguration(10, True)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module

File: software/examples/delphi/ExampleCallback.pas
program ExampleCallback;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletMultiTouchV2;

type
  TExample = class
  private
    ipcon: TIPConnection;
    mt: TBrickletMultiTouchV2;
  public
    procedure TouchStateCB(sender: TBrickletMultiTouchV2; const state: word);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'XYZ'; { Change XYZ to the UID of your Multi Touch Bricklet 2.0 }

var
  e: TExample;

{ Callback procedure for touch state callback }
procedure TExample.TouchStateCB(sender: TBrickletMultiTouchV2; const state: word);
begin
  WriteLn(Format('Touch State: %d', [state]));
end;

procedure TExample.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  mt := TBrickletMultiTouchV2.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Register touch state callback to procedure TouchStateCB }
  mt.OnTouchState := {$ifdef FPC}@{$endif}TouchStateCB;

  { Set period for touch state callback to 0.01s (10ms) }
  mt.SetTouchStateCallbackConfiguration(10, true);

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.
